### PR TITLE
nixos/lib/make-disk-image: do not compile a full arch QEMU to convert images

### PR DESCRIPTION
**Pull request by RaitoBezarius**
###### Description of changes

We were using `qemu` instead of `qemu_kvm` or `qemu-utils`, which is a shame because it will compile a full blown QEMU for that.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Review by alyssais — APPROVED** [review]

File: nixos/lib/make-single-disk-zfs-image.nix [hunk 1]
```diff
@@ -244,7 +244,7 @@ let
             ${if formatOpt == "raw" then ''
             mv $rootDiskImage $out/${rootFilename}
           '' else ''
-            ${pkgs.qemu}/bin/qemu-img convert -f raw -O ${formatOpt} ${compress} $rootDiskImage $out/${rootFilename}
+            ${pkgs.qemu_kvm}/bin/qemu-img convert -f raw -O ${formatOpt} ${compress} $rootDiskImage $out/${rootFilename}
```
Comment (alyssais) [inline nixos/lib/make-single-disk-zfs-image.nix:247]: It looks like runInLinuxVM actually uses `qemu`.  Maybe that should be fixed?

**Review by alyssais — CHANGES_REQUESTED** [review]

File: nixos/maintainers/scripts/openstack/openstack-image-zfs.nix [hunk 2]
```diff
@@ -85,7 +85,7 @@ in
         ${pkgs.jq}/bin/jq -n \
           --arg system_label ${lib.escapeShellArg config.system.nixos.label} \
           --arg system ${lib.escapeShellArg pkgs.stdenv.hostPlatform.system} \
-          --arg root_logical_bytes "$(${pkgs.qemu}/bin/qemu-img info --output json "$rootDisk" | ${pkgs.jq}/bin/jq '."virtual-size"')" \
+          --arg root_logical_bytes "$(${pkgs.qemu_kvm}/bin/qemu-img info --output json "$rootDisk" | ${pkgs.jq}/bin/jq '."virtual-size"')" \
```
[the remark this answers is not in the document]
Comment (RaitoBezarius) [reply nixos/maintainers/scripts/openstack/openstack-image-zfs.nix:88]: There's one, we are calling make-single-zfs-disk-image here

[the remark this answers is not in the document]
Comment (RaitoBezarius) [reply nixos/maintainers/scripts/openstack/openstack-image-zfs.nix:88]: we are in "postVM", so surely there must be a VM somewhere

File: nixos/lib/make-single-disk-zfs-image.nix (hunk 1 above)
Comment (RaitoBezarius) [reply nixos/lib/make-single-disk-zfs-image.nix:247]: to the best of my knowledge it uses `    ${qemu-common.qemuBinary qemu} \`
and `qemu = buildPackages.qemu_kvm` if I'm not wrong

Comment (RaitoBezarius) [reply nixos/lib/make-single-disk-zfs-image.nix:247]: therefore qemu_kvm is good here too

File: nixos/tests/systemd-repart.nix [hunk 3]
```diff
@@ -21,7 +21,7 @@ let
     shutil.copyfile("${machine.system.build.diskImage}/nixos.img", tmp_disk_image.name)
 
     subprocess.run([
-      "${pkgs.qemu}/bin/qemu-img",
+      "${pkgs.qemu_kvm}/bin/qemu-img",
```
[the remark this answers is not in the document]
Comment (RaitoBezarius) [reply nixos/tests/systemd-repart.nix:24]: done

File: nixos/lib/make-single-disk-zfs-image.nix (hunk 1 above)
Comment (alyssais) [reply nixos/lib/make-single-disk-zfs-image.nix:247]: You're right, I missed the `qemu` binding.  That's confusing.

File: nixos/maintainers/scripts/openstack/openstack-image-zfs.nix (hunk 2 above)
[the remark this answers is not in the document]
Comment (alyssais) [reply nixos/maintainers/scripts/openstack/openstack-image-zfs.nix:88]: Yep, you're right.

**Review by alyssais — APPROVED** [review]

**Pull request merged**